lib/api/apiUtils/bucket/bucketCreation.js

```diff
@@ -136,7 +136,10 @@ export function cleanUpBucket(bucketMD, canonicalID, log, callback) {
 export function createBucket(authInfo, bucketName, headers,
     locationConstraint, log, cb) {
     log.trace('Creating bucket');
-    assert.strictEqual(typeof bucketName, 'string');
+    if (typeof bucketName !== 'string') {
```

Comment:
That assertion was for type assertions of function parameters (I was not a fan of that). Anyways, shouldn't you be checking `if(bucketName === '')`?